app/views/search/index.js.erb

```diff
@@ -0,0 +1,14 @@
+$('#search_results').html('<%= escape_javascript(render("search/#{@search_type}")) %>');
+$('#teams_count').html(<%= @teams_count %>);
+$('#namespaces_count').html(<%= @namespaces_count %>);
+$('#repositories_count').html(<%= @repositories_count %>);
+
+function updateCategoryLink(selector) {
+    var new_href = $(selector).attr('href');
+    new_href = new_href.replace(/([\&?]search=)\w*(\&?)/g, '$1<%= @search_query %>$2');
+    $(selector).attr("href", new_href);
+}
```

Comment:
I did this function to update the link (dynamically updates the `search` get parameter) of the search result categories (repositories, namespaces and teams , I think this function won't be used in another context, because of his specificity